CSS adjustments to resize division bar

manolo opened the pull request:
Connected to #456

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vaadin/vaadin-grid/457)
<!-- Reviewable:end -->

jouni commented on vaadin-grid.html:
```diff
@@ -711,26 +711,23 @@
-    background-color: #e3e3e3;
+  :host ::content .vaadin-grid-column-resize-handle:hover,
+  .vaadin-grid-column-resize-handle:hover {
+    border-right: solid 1px #00B4F0;
```
Should use a the themes primary color instead of a hardcoded value.

manolo commented:
Review status: 0 of 7 files reviewed at latest revision, 1 unresolved discussion.

---

*[vaadin-grid.html, line 722 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-grid/457#-KQZfVsBRPV2NUkuiTgb-r1-722:-KQZvdMQtvpxz_zux6OP:b-1dpve7) ([raw file](https://github.com/vaadin/vaadin-grid/blob/d00962810903183c5cec42aeab2985a20e29f1fe/vaadin-grid.html#L722)):*
<details><summary><i>Previously, jouni (Jouni Koivuviita) wrote…</i></summary>
> Should use a the themes primary color instead of a hardcoded value.
</details>
At the same time I replace all occurrences. And noticed a typo in animations which was causing not being used.

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid/457)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
*[vaadin-grid.html, line 340 \[r2\]](https://reviewable.io:443/reviews/vaadin/vaadin-grid/457#-KQ_2Mxfxe6p_mEnLCqj:-KQ_2MxgNYxmxg6v297z:bdu4mzt) ([raw file](https://github.com/vaadin/vaadin-grid/blob/88fedb2d0211afe667ff43d6132817060545693a/vaadin-grid.html#L340)):*
> ```HTML
> 
>   :focus .vaadin-grid-row-focused {
>     -webkit-animation: vaadin-grid-row-focused 0.3s ease-in-out;
> ```

Do we really want to enable this animation? The flashing color effect looks strange.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid/457)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
Review status: 0 of 7 files reviewed at latest revision, 3 unresolved discussions.

---

*[vaadin-grid.html, line 722 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-grid/457#-KQZfVsBRPV2NUkuiTgb-r1-722:-KQ_5-W8tTNjK0gckrCe:bbl187f) ([raw file](https://github.com/vaadin/vaadin-grid/blob/d00962810903183c5cec42aeab2985a20e29f1fe/vaadin-grid.html#L722)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary>
> At the same time I replace all occurrences. And noticed a typo in animations which was causing not being used.
> 
> Done.
</details>
#00B4F0 is Vaadin brand theme color which we shouldn't have here. Rather fallback to the #03A9F4 value that's used at least with the focus line.

---

*[vaadin-grid.html, line 723 \[r2\]](https://reviewable.io:443/reviews/vaadin/vaadin-grid/457#-KQ_4g1Bu_wHn9oFCWHk:-KQ_4g1Bu_wHn9oFCWHl:b62ar4z) ([raw file](https://github.com/vaadin/vaadin-grid/blob/88fedb2d0211afe667ff43d6132817060545693a/vaadin-grid.html#L723)):*
> ```HTML
>   .vaadin-grid-column-resize-handle:hover {
>     border-right: solid 1px var(--primary-color, #00B4F0);
>     background-color: #b9e9ff;
> ```

#b9e9ff is yet another color to the grid's pallette, would there be a pre-existing option for this? Should it be configurable also like the border-right color (maybe secondary color)?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid/457)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
*[vaadin-grid.html, line 723 \[r2\]](https://reviewable.io:443/reviews/vaadin/vaadin-grid/457#-KQ_4g1Bu_wHn9oFCWHk:-KQ_9-0fp_oT-MvRjdx7:b-b2zdg6) ([raw file](https://github.com/vaadin/vaadin-grid/blob/88fedb2d0211afe667ff43d6132817060545693a/vaadin-grid.html#L723)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary>
> #b9e9ff is yet another color to the grid's pallette, would there be a pre-existing option for this? Should it be configurable also like the border-right color (maybe secondary color)?
</details>
--primary-color for the border and --light-primary-color for the background as the first options seem to work quite nicely. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid/457)*
<!-- Sent from Reviewable.io -->

manolo commented:
Review status: 0 of 10 files reviewed at latest revision, 3 unresolved discussions.

---

*[vaadin-grid.html, line 722 \[r1\]](https://reviewable.io:443/reviews/vaadin/vaadin-grid/457#-KQZfVsBRPV2NUkuiTgb-r1-722:-KQ_CKrClwoZ7Ip_erX1:b-896fix) ([raw file](https://github.com/vaadin/vaadin-grid/blob/d00962810903183c5cec42aeab2985a20e29f1fe/vaadin-grid.html#L722)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary>
> #00B4F0 is Vaadin brand theme color which we shouldn't have here. Rather fallback to the #03A9F4 value that's used at least with the focus line.
</details>
Done.

---

*[vaadin-grid.html, line 340 \[r2\]](https://reviewable.io:443/reviews/vaadin/vaadin-grid/457#-KQ_2Mxfxe6p_mEnLCqj:-KQ_CH5jzhdQeH5Ub9an:ba3dfob) ([raw file](https://github.com/vaadin/vaadin-grid/blob/88fedb2d0211afe667ff43d6132817060545693a/vaadin-grid.html#L340)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary>
> Do we really want to enable this animation? The flashing color effect looks strange.
</details>
Removed

---

*[vaadin-grid.html, line 723 \[r2\]](https://reviewable.io:443/reviews/vaadin/vaadin-grid/457#-KQ_4g1Bu_wHn9oFCWHk:-KQ_CLoLfg6d7IaA_LTw:b-896fix) ([raw file](https://github.com/vaadin/vaadin-grid/blob/88fedb2d0211afe667ff43d6132817060545693a/vaadin-grid.html#L723)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary>
> --primary-color for the border and --light-primary-color for the background as the first options seem to work quite nicely. 
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid/457)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
Overflow of the resized header cell doesn't look right https://www.dropbox.com/s/ji6onp2mqovbjeg/Screenshot%202016-09-01%2015.23.14.png?dl=0

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid/457#-:-KQ_ibTNui9vMmrFxrQH:bh0xubh)*
<!-- Sent from Reviewable.io -->

manolo commented:
This is a known issue because the separator is absolute positioned to the right with 8px width. There is line in the middle of 1px,  hence there are 4px from the line to the end that is difficult to hide. 
I see two options:
- rollback the line from the middle to the right (using div border instead of centered :after) . That makes that the cursor looks  always displaced some pixels.
- Set a background to the div with the same color of the grid. That makes that the user theming the grid has to style this element as well.

---

Review status: 0 of 10 files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid/457#-:-KQ_xUhWah5wpmPO2Lli:b-trocwo)*
<!-- Sent from Reviewable.io -->

manolo commented:
Finally I got a solution that fixes the problem and also makes headers behave the same that cells in body.

---

Review status: 0 of 10 files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid/457#-:-KQa5uAqXaln4TnQ9a4p:b-jn4fzr)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
*[vaadin-grid.html, line 726 \[r3\]](https://reviewable.io:443/reviews/vaadin/vaadin-grid/457#-KQaNEAkbToW-Wj1_zeJ:-KQaNEAkbToW-Wj1_zeK:b-e4xsjc) ([raw file](https://github.com/vaadin/vaadin-grid/blob/50c479907261c0f7794e24417a0efa39660b7d25/vaadin-grid.html#L726)):*
> ```HTML
>   :host ::content .vaadin-grid-column-resize-handle:active,
>   .vaadin-grid-column-resize-handle:active {
>     transition: all 0.9s ease-in-out;
> ```

Is this transition by the design? Makes the resizing feel a bit sluggish, especially with higher grids (like 10 rows or so).

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid/457)*
<!-- Sent from Reviewable.io -->

manolo commented:
Review status: 0 of 10 files reviewed at latest revision, 4 unresolved discussions.

---

*[vaadin-grid.html, line 726 \[r3\]](https://reviewable.io:443/reviews/vaadin/vaadin-grid/457#-KQaNEAkbToW-Wj1_zeJ:-KQaYaug21sWninZQlvH:b-7o34x9) ([raw file](https://github.com/vaadin/vaadin-grid/blob/50c479907261c0f7794e24417a0efa39660b7d25/vaadin-grid.html#L726)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary>
> Is this transition by the design? Makes the resizing feel a bit sluggish, especially with higher grids (like 10 rows or so).
</details>
It's not in design, to me it looks brusque when the entire line appears on the body. The ideal would be a fade-in, but it cannot since it's the same line  for body and header. I've changed the time to the same value than the focus effect. Now it does not feel sluggish nor brusque in large grids.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid/457)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
Magic numbers shouldn't be used for horizontal centering. Added a commit to fix that.

---

Review status: 0 of 10 files reviewed at latest revision, 6 unresolved discussions.

---

*[vaadin-grid.html, line 707 \[r4\]](https://reviewable.io:443/reviews/vaadin/vaadin-grid/457#-KQdTTpNeB5WN0HqJHNE:-KQdTTpNeB5WN0HqJHNF:b-rc9v0w) ([raw file](https://github.com/vaadin/vaadin-grid/blob/a63dda2a1b9c8e7a4d6dadb86e4d9b0d55452730/vaadin-grid.html#L707)):*
> ```HTML
>     top: 0;
>     height: 100%;
>     width: 7px;
> ```

Is there a reason to have the handle so narrow? Noticed that the framework grid has 36px wide handle which is a more convenient touch target.

---

*[vaadin-grid.html, line 725 \[r4\]](https://reviewable.io:443/reviews/vaadin/vaadin-grid/457#-KQdTELfa4-RaNt33xt9:-KQdTELfa4-RaNt33xtA:bbpkzmc) ([raw file](https://github.com/vaadin/vaadin-grid/blob/a63dda2a1b9c8e7a4d6dadb86e4d9b0d55452730/vaadin-grid.html#L725)):*
> ```HTML
>   }
> 
>   :host ::content .vaadin-grid-column-resize-handle:active,
> ```

:active doesn't work well with the 3000px high element on Android. Use .vaadin-grid-column-resize-handle-dragged selector instead.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid/457#-:-KQdTvycNaefx1T3z3ND:bmpqjyz)*
<!-- Sent from Reviewable.io -->

manolo commented:
Review status: 0 of 10 files reviewed at latest revision, 6 unresolved discussions.

---

*[vaadin-grid.html, line 707 \[r4\]](https://reviewable.io:443/reviews/vaadin/vaadin-grid/457#-KQdTTpNeB5WN0HqJHNE:-KQdmdKh-89RtvjgGx3O:bpi7xw8) ([raw file](https://github.com/vaadin/vaadin-grid/blob/a63dda2a1b9c8e7a4d6dadb86e4d9b0d55452730/vaadin-grid.html#L707)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary>
> Is there a reason to have the handle so narrow? Noticed that the framework grid has 36px wide handle which is a more convenient touch target.
</details>
Anton suggested 8px, I didn't thought so much on this apart from that 1px was so thin. I set 7px so as the 1px line is exactly centered,  we can use any odd number though. Google docs use around 7px (I cannot measure exactly since it's a canvas, but did by comparison). 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid/457)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
*[vaadin-grid.html, line 707 \[r4\]](https://reviewable.io:443/reviews/vaadin/vaadin-grid/457#-KQdTTpNeB5WN0HqJHNE:-KQdqdlAj23Bxj3ucgnU:bw48gyc) ([raw file](https://github.com/vaadin/vaadin-grid/blob/a63dda2a1b9c8e7a4d6dadb86e4d9b0d55452730/vaadin-grid.html#L707)):*
<details><summary><i>Previously, manolo (Manuel Carrasco Moñino) wrote…</i></summary>
> Anton suggested 8px, I didn't thought so much on this apart from that 1px was so thin. I set 7px so as the 1px line is exactly centered,  we can use any odd number though. Google docs use around 7px (I cannot measure exactly since it's a canvas, but did by comparison). 
</details>
Odd number is probably best choice in our case

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid/457)*
<!-- Sent from Reviewable.io -->

manolo commented:
your commit LGTM, thx

---

Review status: 0 of 10 files reviewed at latest revision, 4 unresolved discussions.

---

*[vaadin-grid.html, line 707 \[r4\]](https://reviewable.io:443/reviews/vaadin/vaadin-grid/457#-KQdTTpNeB5WN0HqJHNE:-KQdrdHvJHzwSZF_wNgu:b-zd5kc0) ([raw file](https://github.com/vaadin/vaadin-grid/blob/a63dda2a1b9c8e7a4d6dadb86e4d9b0d55452730/vaadin-grid.html#L707)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary>
> Odd number is probably best choice in our case
</details>
Changed from 7 to 35 to be aligned with FW

---

*[vaadin-grid.html, line 725 \[r4\]](https://reviewable.io:443/reviews/vaadin/vaadin-grid/457#-KQdTELfa4-RaNt33xt9:-KQdrjNdZbQcQvkHacTA:bdanzlf) ([raw file](https://github.com/vaadin/vaadin-grid/blob/a63dda2a1b9c8e7a4d6dadb86e4d9b0d55452730/vaadin-grid.html#L725)):*
<details><summary><i>Previously, tomivirkki (Tomi Virkki) wrote…</i></summary>
> :active doesn't work well with the 3000px high element on Android. Use .vaadin-grid-column-resize-handle-dragged selector instead.
</details>
Added the selectors. Not removed the :active ones because I want the large line being displayed before the user starts dragging.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid/457#-:-KQdrZJK4-wspbeiwRq4:br7rxqz)*
<!-- Sent from Reviewable.io -->

tomivirkki commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 3 of 6 files at r1, 1 of 2 files at r2, 5 of 6 files at r3, 1 of 1 files at r5.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/vaadin-grid/457#-:-KQdv7Mg9aXC2Y1OI3Yb:bnfp4nl)*
<!-- Sent from Reviewable.io -->
